### PR TITLE
Add worker thread for prometheus metrics in fog ledger

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
Previously fog ledger would only report the ledger db metrics on startup and not update the values. Now fog ledger will continue to update the prometheus metrics.

### CODE REVIEW

File: fog/ledger/server/src/key_image_store_server.rs [hunk 1]
```diff
@@ -200,6 +204,11 @@ where
             .expect("failed starting report cache thread"),
         );
 
+        self.metrics_update_thread = Some(
+            MetricsUpdateThread::start(self.ledger.clone(), self.logger.clone())
+                .expect("failed starting metrics update thread"),
+        );
+
```
Comment (nick-mobilecoin) [inline fog/ledger/server/src/key_image_store_server.rs:211]: Added a metrics updater to the stores for symmetry.
One can get some of these values from https://github.com/mobilecoinfoundation/mobilecoin/blob/09062b26226eb6df3cdee082573c4faca2388bb6/fog/ledger/server/src/db_fetcher.rs#L285 but it's a different counter/gauge than the other ones.

File: fog/ledger/server/src/metrics.rs [hunk 2]
```diff
@@ -32,3 +43,74 @@ lazy_static! {
     )
     .expect("metric cannot be created");
 }
+
+pub struct MetricsUpdateThread {
```
Comment (nick-mobilecoin) [inline fog/ledger/server/src/metrics.rs:47]: I'm up for suggestions here, the fog ledger router doesn't have a consistent place that it's checking the block height, so a consistent polling seemed like the best approach.

File: fog/ledger/server/src/key_image_store_server.rs (hunk 1 above)
Comment (nick-mobilecoin) [reply fog/ledger/server/src/key_image_store_server.rs]: re-worked no longer using a dedicated thread for this.

File: fog/ledger/server/src/metrics.rs (hunk 2 above)
Comment (nick-mobilecoin) [reply fog/ledger/server/src/metrics.rs]: Moved logic into the main thread loop

Comment (jgreat) [reply fog/ledger/server/src/metrics.rs]: Our current minimum resolution on monitoring is 30s. If we're polling, we only need to update on that frequency.

Comment (eranrund) [reply fog/ledger/server/src/metrics.rs]: These updates are cheap, they only update an internal counter (that will get polled at the 30s interval)

**Pull request merged**